runtime/src/chplcast.c

```diff
@@ -188,7 +188,16 @@ _define_string_to_float_precise(real, 64, "%lf")
     if (scanningNCounts() && numitems == 3) {                           \
       numitems = 2;                                                     \
     }                                                                   \
-    if (numitems == 1) {                                                \
+    if (numitems == 0) {                                                \
+      if (0 == strncmp(str, "infi", 4)) {                               \
```

Comment:
Given the use of strncmp, will this potentially also cause "infiblah" to become INFINITY (and would we want that for any reason?)?  Is there any reason not to just use strcmp() here?  I don't think there's anything unsafe about it, unlike `strcpy()`, say.

Reply:
Oh, good catch.  Thanks!

Reply:
#27401 will fix this